services/f-droid/f-droid.service.js

```diff
@@ -1,31 +1,43 @@
 'use strict'
 
+const Joi = require('joi')
 const BaseService = require('../base')
+const validate = require('../../lib/validate')
 const { addv: versionText } = require('../../lib/text-formatters')
 const { version: versionColor } = require('../../lib/color-formatters')
-const { InvalidResponse } = require('../errors')
+const { InvalidResponse, InvalidParameter } = require('../errors')
 
 module.exports = class FDroid extends BaseService {
-  async fetch({ appId }) {
+  async fetch({ appId, format }) {
     // currently, we only use the txt format. There are few apps using the yml format.
-    const url = `https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/${appId}.txt`
-    const { buffer } = await this._request({
-      url,
+    const url = `https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/${appId}`
+    const reqOptions = {
+      url: `${url}.${format || 'txt'}`,
       options: {},
       errorMessages: {
         404: 'app not found',
       },
-    })
+    }
+    let buffer
+    try {
+      ;({ buffer } = await this._request(reqOptions))
+    } catch (error) {
+      if (format) {
+        throw error
+      }
+      ;({ buffer } = await this._request({ ...reqOptions, url: `${url}.yml` }))
+    }
     const metadata = buffer.toString()
     // we assume the layout as provided here:
     // https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/axp.tool.apkextractor.txt
-    const positionOfCurrentVersionAtEndOfTheFile = metadata.lastIndexOf(
-      'Current Version:'
-    ) // credits: https://stackoverflow.com/a/11134049
+    const value = (metadata.match(/CurrentVersion:|Current Version:/) || [
+      'Current Version:',
+    ])[0]
+    const positionOfCurrentVersionAtEndOfTheFile = metadata.lastIndexOf(value) // credits: https://stackoverflow.com/a/11134049
```

Comment:
Parsing files manually tends to create extra work and introduce risk. When fetching the yaml version, could you use the yaml parser instead?

It might be easier to split that into separate `fetchText` and `fetchYaml` functions, and to move the retry logic into `handle()`.